src/plugins/index_pattern_management/public/components/create_index_pattern_wizard/create_index_pattern_wizard.tsx

```diff
@@ -244,7 +242,7 @@ export class CreateIndexPatternWizard extends Component<
 
   goToNextFromDataSource = (dataSourceRef: DataSourceRef) => {
     this.setState({ isInitiallyLoadingIndices: true, dataSourceRef }, async () => {
-      await this.fetchData();
+      this.fetchData();
```

Comment:
If await is removed on fetchData() .. this.goToNextStep() will be called before the fetch data is completed

Reply:
Aware of that, however the await will not function anymore after remove the other promise await. Also was setting the loading state first. the state will not change until fetchData finishes. think that's how it ever worked originally -- as it never waits before in componentWillMount